### PR TITLE
Merge kfp-notebook to Elyra repo

### LINKED ISSUE #1132: Refactor kfp-notebook into elyra
Introducing the airflow notebook operator, also introduces a lot of duplicate code, since both the kfp-notebook and airflow-notebook operators use, for the most part, the same image configuration files and scripts. 
This issue is to cover the work regarding moving the kfp-notebook operator in https://github.com/elyra-ai/kfp-notebook into elyra.

- Extend the bootstrapper.py script base class to cover both kfp and airflow operations
- Move and merge existing unit and integration tests
- Separate airflow and kfp runtimes under elyra/pipeline
- Move configuration scripts and files and general housekeeping under etc/

### PR DESCRIPTION
**Pull request by lresende**
Fixes #1132 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by lresende** [comment]
Have submitted a few pipelines to KFP and they seem to be working end-to-end as before the merge.

File: elyra/kfp/operator.py
Comment (kevin-bates) [inline elyra/kfp/operator.py:58]: `ExtendedContainerOp` seems too obvious and not very informative.  How about `ElyraGenericOp` or `ElyraFileOp`.  Since, I suspect, nearly every operator extends `ContainerOp`, I don't think we need to preserve that aspect of things. `ElyraGenericOp` implies it's one of our generic operators, although, per the previous discussion, those may not always be file-based, as they are today - thus, the `ElyraFileOp`.

**Review by kevin-bates — APPROVED** [review]
If this gets rid of kfp-notebook and the associated release-coordination hassles, I'm all for it!  Was the reason we didn't do this initially because we didn't want to include elyra on the container image?

I just had the one comment about the name - `ExtendedContainerOp`, but am approving assuming that can be updated.

File: elyra/pipeline/tests/test_processor_airflow.py
Comment (akchinSTC) [inline elyra/pipeline/tests/test_processor_airflow.py:188]:
```suggestion
                if "notebook_op_" + node['id'].replace("-", "_") + " = NotebookOp(" in file_as_lines[i]:
```

File: elyra/templates/airflow/airflow_template.jinja2
Comment (akchinSTC) [inline elyra/templates/airflow/airflow_template.jinja2:5]:
```suggestion
from airflow_notebook.pipeline import NotebookOp
```

File: elyra/templates/airflow/airflow_template.jinja2
Comment (akchinSTC) [inline elyra/templates/airflow/airflow_template.jinja2:48]:
```suggestion
op_{{ operation.id|regex_replace }} = NotebookOp(name='{{ operation.notebook|regex_replace }}',
```

**Review by akchinSTC — CHANGES_REQUESTED** [review]
lets revert these back to the original class name, and we can update the airflow notebook to a better name the next time we need to cut a release.

File: elyra/kfp/operator.py
Comment (akchinSTC) [reply elyra/kfp/operator.py:58]: +1, a little too on the nose

Comment (lresende) [reply elyra/kfp/operator.py:58]: I am not sure I wanted to use Elyra on the name of the op, would ExtendedFileOp or just FileOp be ok?

**Comment by kevin-bates** [comment]
This looks similar to #1132.  @akchinSTC - does this PR sufficiently address the issue?

File: elyra/kfp/operator.py
Comment (kevin-bates) [reply elyra/kfp/operator.py:58]: I understand not wanting `Elyra` in the op name.  `FileOp` seems fine.  @ptitzler had suggested `RunFileOp` in a sidebar and I like that too and that's what happens.  I guess in that sense `ExecuteFileOp` also works.  I don't think `Extended` fits this context.

**Comment by akchinSTC** [comment]
looks like it should satisfy the issue, the airflow uses its own copy of the boostrapper script but as far as I can tell there isnt any different between the two, dont think there should be considering the script is running inside an isolation container/pod and doesnt contain any runtime specific properties.
we should update the airflow-notebook as well next go around to pull scripts from elyra as well. I think we might be able to drop the airflow package req as well

**Review by akchinSTC — APPROVED** [review]

**Comment by kevin-bates** [comment]
The final changes look good - this will be nice!

**Pull request merged**
